crates/ibc-elc-client-cw/src/context.rs

```diff
@@ -0,0 +1,351 @@
+            Some(checksum) => Ok(checksum.clone()),
+            None => {
+                let value = self.get_prefixed(CLIENT_STATE.as_bytes())?;
+                let any_wasm_client_state = Any::decode(value.as_slice())?;
+                let wasm_client_state: WasmClientState = any_wasm_client_state.try_into()?;
+                Ok(wasm_client_state.checksum.into())
+            }
+        }
+    }
+
+    pub fn set_subject_prefix(&mut self) {
+        self.migration_prefix = Some(SUBJECT_PREFIX);
+    }
+
+    pub fn set_substitute_prefix(&mut self) {
+        self.migration_prefix = Some(SUBSTITUTE_PREFIX);
+    }
+
+    pub fn prefixed_key(&self, key: impl AsRef<[u8]>) -> Vec<u8> {
+        let mut prefixed_key = Vec::new();
+        prefixed_key.extend_from_slice(self.migration_prefix.unwrap_or(b""));
+        prefixed_key.extend_from_slice(key.as_ref());
+        prefixed_key
+    }
+
+    pub fn get_prefixed(&self, key: impl AsRef<[u8]>) -> Result<Vec<u8>, ContractError> {
+        let prefixed_key = self.prefixed_key(key);
+        self.storage_ref()
+            .get(&prefixed_key)
+            .ok_or(ContractError::generic(format!(
+                "value not found in storage: key={}",
+                String::from_utf8(prefixed_key).unwrap(),
+            )))
+    }
+
+    pub fn set_prefixed(&mut self, key: impl AsRef<[u8]>, value: impl AsRef<[u8]>) {
+        let prefixed_key = self.prefixed_key(key);
+        self.storage_mut().set(&prefixed_key, value.as_ref());
+    }
+
+    pub fn remove_prefixed(&mut self, key: &[u8]) {
+        let prefixed_key = self.prefixed_key(key);
+        self.storage_mut().remove(&prefixed_key);
+    }
+
+    pub fn host_height(&self) -> Height {
+        Height::new(0, self.env.block.height)
+    }
+}
+
+impl<'a, C: CustomQuery> KVStore for Context<'a, C> {
+    fn set(&mut self, key: Vec<u8>, value: Vec<u8>) {
+        self.set_prefixed(key, value)
+    }
+
+    fn get(&self, key: &[u8]) -> Option<Vec<u8>> {
+        self.get_prefixed(key).ok()
+    }
+
+    fn remove(&mut self, key: &[u8]) {
+        self.remove_prefixed(key)
+    }
+}
+
+impl<'a, C: CustomQuery> HostContext for Context<'a, C> {
+    fn host_timestamp(&self) -> Time {
+        Time::from_unix_timestamp_nanos(self.env.block.time.nanos() as u128).unwrap()
+    }
+}
+
+impl<'a, C: CustomQuery> ClientReader for Context<'a, C> {
+    fn client_exists(&self, _client_id: &ClientId) -> bool {
+        unimplemented!();
+    }
+
+    fn client_type(&self, _client_id: &ClientId) -> Result<String, LightError> {
+        unimplemented!();
+    }
+
+    fn client_state(&self, client_id: &ClientId) -> Result<Any, LightError> {
+        let value = self
+            .get(CLIENT_STATE.as_bytes())
+            .ok_or(LightError::client_state_not_found(client_id.clone()))?;
+        let any_wasm_client_state = Any::decode(value.as_slice())
+            .map_err(|_| LightError::client_state_not_found(client_id.clone()))?;
```

Comment:
L180-185: These error reasons are not actually "client state not found".

Reply:
fixed: https://github.com/datachainlab/ibc-elc-client-cw/pull/1/commits/7bf4fb0b89646fca174e392bab26f46548329192